lib/lift_wing_api.rb

```diff
@@ -72,36 +101,103 @@ def lift_wing_server
     connection
   end
 
-  # To make migration from ORES to LiftWing easier, we want responses to be in the same format,
-  # including error responses.
-  # For a deleted revision, ORES returns something like this:
-  # {"enwiki"=>{"scores"=>{"708326238"=>{"articlequality"=>{"error"=>
-  #   {"message"=>"TextDeleted: Text deleted (datasource.revision.text)",
-  #    "type"=>"TextDeleted"}}}}}}
-  # Lift Wing just returns something like:
-  # {"error"=>
-  #  "Missing resource for rev-id 708326238: TextDeleted: Text deleted (datasource.revision.text)"}
-  ERROR_TYPE_MATCHER = Regexp.union DELETED_REVISION_ERRORS
-
-  def equivalent_ores_error_response(rev_id, error_response)
-    message = error_response['error']
-    type = message[ERROR_TYPE_MATCHER]
-
-    { @project_code =>
-      { 'scores' => { rev_id.to_s => { @project_quality_model => {
-        'error' => { 'message' => error_response['error'], 'type' => type }
-        } } } } }
-  end
-
   # TODO: monitor production for errors, understand them, put benign ones here
   TYPICAL_ERRORS = [].freeze
 
   def log_error_batch(rev_ids)
     return if @errors.empty?
 
     log_error(@errors.first, update_service: @update_service,
```

Comment:
I understand that we always log the first error, even when there might be multiple distinct errors. I guess we work under the assumption that if several requests fail, it's likely for the same reason, which may hold true in most cases. However, I suggest resetting the `@errors` array in `get_revision_data`. Without this adjustment, if there are two batches of requests, each with a single request failing, we would log the first error twice and miss the second one entirely.

Reply:
Done on cde5e1d182f2b302b043cd8a6b538d240fc138da